packages/backend/src/services/OrganizationService/OrganizationService.ts

```diff
@@ -752,4 +778,157 @@ export class OrganizationService extends BaseService {
 
         return palette;
     }
+
+    async initializeProject() {
+        // No permissions check here, there are no users yet
+        // No initial setup, we skip this step
+        if (!this.lightdashConfig.initialSetup) return;
+        try {
+            const setup = this.lightdashConfig.initialSetup;
+            // If no project is set, we can create a new one using environment variables
+            const hasOrgs = await this.organizationModel.hasOrgs();
+
+            if (hasOrgs) {
+                this.logger.debug(
+                    `Initial setup: There is already an organization, we skip this initial setup`,
+                );
+                // There is already an organization, we skip this step
+                return;
+            }
+            const hasAnyProjects = await this.projectModel.hasAnyProjects();
+
+            if (hasAnyProjects) {
+                // This should not happen, since we can't have projects without orgs
+                throw new UnexpectedServerError(
+                    `Initial setup: There is already a project, we skip this initial setup`,
+                );
+            }
+
+            // No organization and no projects, we can create a new one
+            // using the initial setup config
+            this.logger.debug(
+                `Initial setup: Creating organization "${setup.organization.name}"`,
+            );
+
+            const organization = await this.organizationModel.create({
```

Comment:
This would have less logic if there was `organizationModel.createWithEmailDomains` or it had optional fields